Update release notes for 0.6.0 (#5195)
hyperledger-labs/splice: docs/src/release_notes.rst

.. _release_notes:

.. release-notes:: 0.6.0

    - Canton

>         - Upgrade to Canton 3.5.0. Note that unlike the upgrade to 3.4.0 this is still just applied like any other upgrade through a helm upgrade or upgrade of the compose files.

Reviewer: ```suggestion
        - Upgrade to Canton 3.5.0. Note that unlike previous bigger upgrades (such as to 3.4.0) this upgrade is applied in the regular (asynchronous) manner through, respectively, a helm upgrade or an upgrade of the compose files.
```

Reviewer: And actually - don't we upgrade to 3.5.**1**? Shall we mention somewhere that this is expected, in case someone wonders about it?

Author: I've conveniently sidestepped this and called it 3.5

          Reference `UNRELEASED.md <https://github.com/digital-asset/canton/blob/release-line-3.5/UNRELEASED.md>`_ for release notes.

>           More polished release notes will be made available later.

Reviewer: When is later?

Author: when Canton tells me they're happy

    - SV App

>         - Package versions newer than the version specified in the AmuletRules configuration are now automatically unvetted by the SV app after a successful downgrade vote.

Reviewer: ```suggestion
        - Package versions *newer* than the version specified in the AmuletRules configuration are now automatically unvetted by the SV app after a successful downgrade vote.
```

Nit; not sure if others will read this as "older" first and get confused, but I did.

          guaranteed to be consistent across all instances of Scan, therefore is suitable for BFT reads.
          The update ID will be omitted for contracts created in a prior migration ID, or potentially in the
          future in extreme cases of disaster recovery.

    - Validator

        - The HTTP client used by the CN apps (Validator, Scan, SV, Wallet) now honours the
          standard ``http.nonProxyHosts`` Java system property to bypass a configured HTTP
          forward proxy for specific hosts. This matches the JDK's own default ``ProxySelector``
          behaviour, so the same property also applies to other JVM egress components.
          See :ref:`validator-http-proxy-compose` and :ref:`validator-http-proxy-helm` for configuration examples.

    - LocalNet

        - Added support for configuring the protocol version used in LocalNet.

    - Participant

        - Set ``commitment-use-db-snapshot-for-participant-lookup = true`` by default. If you manually set this, you can
          remove your overwrite.

    - Daml

        - Remove use of rollback nodes to support protocol version 35

>             - Daml

Reviewer: Daml under Daml is a bit odd, no?

Author: fixed

                - ``AmuletRules``. Replace ``InvalidTransfer`` exceptions with ``failWithStatus``:

                    - ``ITR_InsufficientFunds`` → ``FailureStatus`` with error_id = ``splice.lfdecentralizedtrust.org/insufficient-funds``
                    - ``ITR_UnknownSynchronizer`` → ``FailureStatus`` with error_id = ``splice.lfdecentralizedtrust.org/unknown-synchronizer``

>                     - ``ITR_InsufficientTopupAmount`` → `FailureStatus` with error_id = ``splice.lfdecentralizedtrust.org/insufficient-topup-amount``

Reviewer: ```suggestion
                    - ``ITR_InsufficientTopupAmount`` → ``FailureStatus`` with error_id = ``splice.lfdecentralizedtrust.org/insufficient-topup-amount``
```

                    - ``ITR_Other ("More than the maximum number of outputs")`` → ``FailureStatus`` with error_id = ``splice.lfdecentralizedtrust.org/maximum-outputs-exceeded``

                - ``WalletAppInstall_ExecuteBatch``. No longer catches exceptions and returns them as ``AmuletOperationOutcome`` / ``COO_Error``. Instead, the transaction is aborted.

                - ``TransferCommand_Send``. No longer catches exceptions and returns them as ``TransferCommandResultFailure``. Instead, the transaction is aborted without merging inputs.

                - ``DsoRules_CloseVoteRequest`` no longer catches exceptions. Previously, it would close the vote request with outcome ``VRO_AcceptedButActionFailed``.
                  The transaction will now abort on failure.

        - The Daml compiler used for the splice DARs has been upgraded to 3.4.11.

        - The Daml changes require an upgrade to these versions:

            ================== =======
            name               version
            ================== =======
            amulet             0.1.18
            amuletNameService  0.1.19
            dsoGovernance      0.1.24
            validatorLifecycle 0.1.7

>             wallet             0.1.19

Reviewer: I checked these and they look right to me.

.. _release_notes:

.. release-notes:: 0.6.0

    - Canton

        - Upgrade to Canton 3.5. Note that unlike the upgrade to 3.4 this is still just applied like any other upgrade through a helm upgrade or upgrade of the compose files.
          Reference `UNRELEASED.md <https://github.com/digital-asset/canton/blob/release-line-3.5/UNRELEASED.md>`_ for release notes.
          More polished release notes will be made available later.

    - SV App

        - Package versions *newer* than the version specified in the AmuletRules configuration are now automatically unvetted by the SV app after a successful downgrade vote.

    - SV UI

>         - Various improvements to the SV UI

Reviewer 2: Not sure this is very informative. I'd either provide more detail or drop it.

Author: can't win, @martinflorian-da asked me to add it you're asking me to remove it. Removed it again and I'll let you two fight it out.

Reviewer: I don't really care :) Just seemed noteworthy that we're improving things there.

Reviewer 2: I do tend to agree that we should lower the bar of what we're reporting there, and explicitly list the improvements that are being made. It is definitely user-facing changes.
@pawelperek-da @ray-roestenburg-da

Reviewer 3: ack
